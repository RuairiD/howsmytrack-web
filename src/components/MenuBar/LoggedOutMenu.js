import React from "react";

import { Menu, Icon, Typography } from "antd";
import { A, Span } from "lemon-reset";

const LoggedOutMenu = ({ onMenuClick }) => (
    <Menu
        theme="light"
        onClick={onMenuClick}
        style={{
            borderRight: 0,
        }}
    >
        <Menu.Item disabled>
            <Typography.Text>Welcome!</Typography.Text>
        </Menu.Item>
        <Menu.Item className="d-flex" key="login">
            <Icon className="align-middle my-auto" type="user" />
            <Span>Sign In</Span>
        </Menu.Item>
        <Menu.Item className="d-flex" key="register">
            <Icon className="align-middle my-auto" type="user-add" />
            <Span>Register</Span>
        </Menu.Item>
        <Menu.Item key="faq">
            <A href="/faq" className="d-flex">
                <Icon className="align-middle my-auto" type="question-circle" />
                <Span>FAQ</Span>
            </A>
        </Menu.Item>
    </Menu>
);

export default LoggedOutMenu;
