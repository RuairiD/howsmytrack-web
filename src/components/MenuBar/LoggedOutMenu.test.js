import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import LoggedOutMenu from "./LoggedOutMenu";

describe("LoggedOutMenu", () => {
    it("renders menu for logged out users with Sign In, Register and FAQ options", () => {
        const wrapper = shallow(
            <LoggedOutMenu
                onMenuClick={() => {}}
            />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
