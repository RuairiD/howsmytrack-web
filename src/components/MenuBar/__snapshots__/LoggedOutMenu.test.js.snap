// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoggedOutMenu renders menu for logged out users with Sign In, Register and FAQ options 1`] = `
<Menu
  onClick={[Function]}
  style={
    Object {
      "borderRight": 0,
    }
  }
  theme="light"
>
  <MenuItem
    disabled={true}
  >
    <Text>
      Welcome!
    </Text>
  </MenuItem>
  <MenuItem
    className="d-flex"
    key="login"
  >
    <Icon
      className="align-middle my-auto"
      type="user"
    />
    <Span
      className=""
    >
      Sign In
    </Span>
  </MenuItem>
  <MenuItem
    className="d-flex"
    key="register"
  >
    <Icon
      className="align-middle my-auto"
      type="user-add"
    />
    <Span
      className=""
    >
      Register
    </Span>
  </MenuItem>
  <MenuItem
    key="faq"
  >
    <A
      className="d-flex"
      href="/faq"
    >
      <Icon
        className="align-middle my-auto"
        type="question-circle"
      />
      <Span
        className=""
      >
        FAQ
      </Span>
    </A>
  </MenuItem>
</Menu>
`;
